MapboxNavigation/RouteMapViewController.swift

```diff
@@ -83,42 +78,25 @@ class RouteMapViewController: UIViewController, PulleyPrimaryContentControllerDe
         } else {
             setDefaultCamera(animated: false)
         }
-
-        UIDevice.current.addObserver(self, forKeyPath: "batteryState", options: .initial, context: nil)
     }
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
 
         mapView.setUserLocationVerticalAlignment(.bottom, animated: false)
         mapView.setUserTrackingMode(.followWithCourse, animated: false)
-
-        if simulatesLocationUpdates {
-            mapView.locationManager.stopUpdatingLocation()
-            mapView.locationManager.stopUpdatingHeading()
-        }
+        
+        mapView.locationManager.stopUpdatingLocation()
+        mapView.locationManager.stopUpdatingHeading()
```

Comment:
These should move to `viewWillAppear` to fix animation issues. More here: https://github.com/mapbox/mapbox-navigation-ios/issues/303